output.c

```diff
@@ -318,7 +324,10 @@ Output_t* Output_Create(bool IsNULL,
 	u32 CPUCnt 	= 0;
 	for (int i=0; i < 32; i++)
 	{
-		pthread_create(&O->PushThread[i], NULL, Output_Worker, (void*)O); 
+		ThreadData_t *T = malloc(sizeof(ThreadData_t));
+		T->Out = O;
```

Comment:
T->Out = NULL;

Reply:
We need to have "Output_t / O" structure access in Output_Worker....so either we can set inside new allocated structure (T) or we have to make "O" (static) globally accessible.

Reply:
As mentioned in chat, its more readable to assign a pointer NULL instead of 0. As assigning 0 kind of implies the field is a numeric value.

Reply:
It's not "0"...it's "O" (Output_t* O = ...)

Reply:
oh my bad